pkg/apis/datadependency/v1alpha1/types.go

```diff
@@ -0,0 +1,189 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+
+// DataSource represents a cached query result for data sources in the federated environment.
+// It is a cluster-scoped resource.
+type DataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataSourceSpec   `json:"spec,omitempty"`
+	Status DataSourceStatus `json:"status,omitempty"`
+}
+
+// DataSourceSpec defines the desired state of DataSource.
+type DataSourceSpec struct {
+	// System specifies the underlying data system.
+	// This provides context for the 'name' and 'attributes' fields.
+	// e.g., "hive", "s3", "hdfs".
+	// +required
+	System string `json:"system"`
+
+	// Type specifies the category of the data source within the system.
+	// e.g., for system="hive", type could be "table", "view".
+	// e.g., for system="s3", type could be "bucket", "object", "prefix".
+	// +required
+	Type string `json:"type"`
+
+	// Name is the identifier of the data source, its format is interpreted
+	// in the context of the 'system' and 'type'.
+	// +required
+	Name string `json:"name"`
+
+	// Locality defines which clusters this data source is available on.
+	// +required
+	Locality *DataSourceLocality `json:"locality"`
+
+	// Attributes provides extra, non-identifying metadata.
+	// +optional
+	Attributes map[string]string `json:"attributes,omitempty"`
+
+	// ReclaimPolicy defines what happens to this DataSource when its last bound DataSourceClaim is deleted.
+	// Defaults to "Retain".
+	// +optional
+	ReclaimPolicy DataSourceReclaimPolicy `json:"reclaimPolicy,omitempty"`
+}
+
+// DataSourceLocality specifies the cached location information of the data source.
+type DataSourceLocality struct {
+	// ClusterNames is a list of cluster names where the cached data source information indicates availability.
+	// This provides a simple and direct way to specify cached data source location
+	// without interfering with user-defined ResourceBinding cluster affinity.
+	// +required
+	ClusterNames []string `json:"clusterNames"`
+}
+
+// DataSourceStatus defines the observed state of DataSource.
+type DataSourceStatus struct {
+	// ClaimRefs is a list of references to DataSourceClaims that are bound to this DataSource.
+	// The presence of items in this list indicates the DataSource is in use.
+	// +optional
+	ClaimRefs []corev1.ObjectReference `json:"claimRefs,omitempty"`
+
+	// BoundClaims counts the number of DataSourceClaims currently bound to this DataSource.
+	// This provides a quick summary of its usage.
+	// +optional
+	BoundClaims int32 `json:"boundClaims,omitempty"`
+
+	// Conditions store the latest available observations of the DataSource's state.
+	// e.g., {type: "Ready", status: "True"}, {type: "Provisioned", status: "True"}
```

Comment:
Do we have these conditions? I didn't find these type of conditions in volcano-global PR